torch/_higher_order_ops/while_loop.py

```diff
@@ -97,8 +97,27 @@ def body_fn(iter, x):
         - 'while_loop' only supports **inference** right now. Autograd will be supported in the future.
 
     """
-    assert isinstance(operands, tuple)
-    return while_loop_op(cond_fn, body_fn, operands)
+    if torch._dynamo.is_compiling():
+        return while_loop_op(cond_fn, body_fn, operands)
+
```

Comment:
We probalby want to validate the input even if we're compiling?

Reply:
I just tried this. The callable builit-in call is not supported in dynamo and we also have redundant check in WhileLoopHigherOrderOp. Can refactor it after supporting callable in dynamo. Basically, using your previous idea of inlining into python input checks in dynamo.